app/Livewire/TrialBalance/AddTrialBalance.php

```diff
@@ -147,9 +151,94 @@ public function update()
         $this->redirect('/trial-balances', navigate: true);
     }
 
+    public function importFromGL()
+    {
+        // load excel template
+        if ($this->tbType) {
+            $tbExportFormat = 'TB_' . strtoupper($this->tbType);
+        } else {
+            $tbExportFormat = "TB_PRE";
+        }
+        $filePath = 'public/uploads/' . $tbExportFormat . '.xlsx';
+        $newFilePath = 'uploads/' . $tbExportFormat . '.xlsx';
+        Storage::copy($filePath, $newFilePath);
+        $spreadsheet = IOFactory::load(storage_path('app/' . $newFilePath));
+
+        // load export config
+        $tbExportConfig = DB::select('SELECT template FROM report_templates WHERE template_name = ?', [$tbExportFormat]);
+        $jsonConfig = array_column($tbExportConfig, 'template')[0];
+        $jsonConfig = json_decode($jsonConfig, true);
+
+        // query data from GL
+        $queryMonth = date('m', strtotime($this->tbDate));
+        $queryYear = date('Y', strtotime($this->tbDate));
+        $this->interimPeriod = trim($this->interimPeriod);
+        if ($this->interimPeriod == 'Quarterly') {
+            $months = [];
+            switch ($this->quarter) {
+                case 'Q1':
+                    $months = ['01', '02', '03'];
```

Comment:
Do you have a copy of their data types sa schema nila? I think number kasi yung sa database nila e, will this work parin kaya?
![image](https://github.com/chaaals/se-financial-reporting/assets/70326902/a704d022-44bc-4644-b144-0d82a1e115ec)

Reply:
yep this won't work anymore. it is still a string tho. i saw their uml. i only assumed it had 0 since ours had 0